contrib/buildrefactor/src/python/pants/contrib/buildrefactor/peek.py

```diff
@@ -0,0 +1,47 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.task.task import Task
+from pants.contrib.buildrefactor.buildozer import Buildozer
+
+
+class Peek(Task):
+  """Peek a build file for a specified target
+
+    line-number: optional flag to print the starting and ending line numbers of the target
+
+    Example:
+      $./pants peek --line-number testprojects/tests/java/org/pantsbuild/testproject/dummies:passing_target
+  """
+
+  @classmethod
+  def register_options(cls, register):
+    super(Peek, cls).register_options(register)
+
+    register('--line-number', help='Prints the starting line number of the named target.', type=bool, default=False)
+
+  def __init__(self, *args, **kwargs):
+    super(Peek, self).__init__(*args, **kwargs)
+
+    self.options = self.get_options()
+
+  def execute(self):
+    self.print_name()
+
+  def print_name(self):
+    for root in self.context.target_roots:
+      address_spec = root.address.spec
+
+    print('\n')
+    if self.options.line_number:
+      print('\n\'{}\' starts on line number:'.format(root.name))
+      Buildozer.execute_binary('print startline', address_spec, suppress_warnings=True)
+      print('\n\'{}\' ends on line number:'.format(root.name))
+      Buildozer.execute_binary('print endline', address_spec, suppress_warnings=True)
+    else:
+      Buildozer.execute_binary('print name', address_spec, suppress_warnings=True)
```

Comment:
The buildozer arg here should be `print rule`
```
$ buildozer 'print rule' examples/tests/java/org/pantsbuild/example/hello/greet:greet
junit_tests(
    dependencies = [
        "examples/src/java/org/pantsbuild/example/hello/greet",
        "examples/src/resources/org/pantsbuild/example/hello",
    ],
)
```
so the method should be renamed to `print_target` as well

Reply:
Implemented as `ConsoleTask`, updated to `'print rule'`, and renamed to `'print_target'` on branch peek-console-task. If the changes look good I'll either update this PR or open a new 'Print Target' PR.

Reply:
Updating this one would be good. Thanks.

Also make sure there is no `print` function left in the code, because all the console output should be the return value.

Reply:
Updated to use `yield` instead of `print`